app/sweepers/collection_sweeper.rb

```diff
@@ -46,12 +46,20 @@ def get_collections_from_record(record)
   def expire_collection_cache_for(record)
     collections = get_collections_from_record(record)
     collections.each do |collection|
-
-      # expire the collection blurb and dashboard and profile
-      ActionController::Base.new.expire_fragment("collection-blurb-#{collection.id}-v3")
-      ActionController::Base.new.expire_fragment("collection-profile-#{collection.id}")
-
+      CollectionSweeper.expire_collection_blurb_and_profile(collection)
     end
   end
 
+  # Expire the collection blurb and profile
+  def self.expire_collection_blurb_and_profile(collection)
+    # Expire all versions of the blurb: whether the new search is enabled or
```

Comment:
Do we want a TODO for removing the old-search/new-search consideration when ES is full switched over?

Reply:
Currently, it also changes the view based on whether `@admin_settings.disable_filtering?` is on or off, since disabling filtering also disables the ES bookmarkable-based view. I'm not sure whether it's worth keeping separate caches just for that, so I added a TODO to consider it later.

Reply:
I live in hope that we might round these up and action them some day so thanks for adding the explanation for future reference.

Reply:
iirc, we've been muttering to ourselves about getting rid of that admin setting